### ISSUE TITLE
HTTP requests are not upgraded to HTTPS

### ISSUE THREAD

**Issue by tmattio** [ocaml/ocaml.org#81]
For obvious security reasons, we should redirect HTTP requests to HTTPS.

The HTTPS handling is done with a proxy/load balancer at the moment, so the redirection should probably be handled there, but we might also want to expose the server directly: Dream should handle HTTPS nicely.

**Comment by patricoferris** [comment]
Didn't see this issue, I think it is likely a Caddy server configuration problem in V3 (http://ocaml.org redirects). 

Waiting on some feedback here and then will try to fix: https://github.com/ocurrent/ocurrent-deployer/pull/76#issuecomment-915309529

**Comment by avsm** [comment]
It would be nice to have the Dream server directly exposed. All that's missing is automatic LetsEncrypt certificate fetching which Caddy does. What are the chances of getting that working with a bit of Dream middleware? :-)

**Comment by tmattio** [comment]
> What are the chances of getting that working with a bit of Dream middleware? :-)

Doesn't seem too complicated, assuming that Dream supports upgrading connections from HTTP to HTTPS, which I think it does. cc @aantron

I'll look into it. Do you know where the certificates are downloaded from atm? How are they protected?

**Comment by aantron** [comment]
Dream currently supports establishing an HTTPS connection.

To redirect from HTTP to HTTPS, you can currently issue an ordinary redirect to the client from a helper Dream server running in-process on port 80 (or other port, if using a proxy) and not using HTTP, to a main Dream server running in the same process on port 443 and using HTTPS. This should probably be wrapped up in some kind of neat helper.

Dream doesn't have nice Let's Encrypt support, but that is planned :)

**Comment by tmattio** [comment]
Closing in favor of #231
